src/src/Helpers/AdafruitGFX_helper.cpp

```diff
@@ -1251,14 +1256,15 @@ uint16_t AdaGFXparseColor(String& s, AdaGFXColorDepth colorDepth) {
  * Convert an RGB565 color (number) to it's name or the #rgb565 hex string, based on depth
  ****************************************************************************************/
 String AdaGFXcolorToString(uint16_t         color,
-                           AdaGFXColorDepth colorDepth) {
+                           AdaGFXColorDepth colorDepth,
```

Comment:
Maybe you can split this function into 2 functions (1 can be only defined in the .cpp to make sure it will not be called from elsewhere)

The extensive switch part only returns a flash string, but does have a lot of return paths where it actually returns a `String` instead of a flash string.
The reason this `AdaGFXcolorToString` now needs a `String` return type is to handle everything not handled in the switch part. (where `EMPTY_STRING` is used you may need to use `F("")` )
The reason to do so is to reduce the build size quite a bit as every `return F("...");` statement is now turned into `return String(F("..."));`, which may add up in the end.

Reply:
I did apply this change, and though not enormous, it still shaved off 52 bytes on an ESP32 build, and even _144_ bytes on an ESP8266 build 👀 so that's going to stay 😅

Reply:
When I did discover this, I went through lots and lots of enum-to-string functions and function calls with String arguments and managed to shave off over 70k of the ESP32 max build.
It really matters to have less implicit conversions to a String type (thus from a type matching one of the many constructors of String)
So any function returning a String with lots of return paths can benefit from this.